documentation/general/contributing.md

```diff
@@ -86,6 +86,11 @@ issue be assigned to you when you're ready to work on it.
 - [Good first issues](https://github.com/WordPress/openverse/issues?q=is:issue+is:open+sort:updated-desc+label:%22good+first+issue%22)
 - [Issues wanting help](https://github.com/WordPress/openverse/issues?q=is:issue+is:open+sort:updated-desc+label:%22help+wanted%22)
 
+It is recommended that first-time contributors begin with a "good first issue".
+For more information on how Openverse maintainers think about "good first
+issues", please see
+[our additional documentation](/meta/contribution/good_first_and_help_wanted_issues.md).
```

Comment:
The audience of the linked document is a bit mixed, if we include it here. Is it for maintainers or for contributors? Moving the definition of a good first issue to a separate document from the one that covers how maintainers help contributors complete their early contributions (predefs, meta-thoughts about good/bad Openverse domains for good-first-issues, etc).

Some version of the "What makes an issue a good first issue?" section makes sense to have as an all-audience context. How to write the good first issue, on the other hand, is just for maintainers. A longer document might discourage contributors looking for something quick from reading it, potentially having them miss out on the context.

As in the original discussion, I think NeoMutt's separation makes a lot of sense, and has layers to "ease" someone into understanding the different levels of possible contribution.

[Here's the definition of "Easy Coding Tasks"](https://neomutt.org/2016/10/07/news), linked from [the "Newbie Tutorial"](https://neomutt.org/dev/newbie-tutorial#coding). Ignoring the list of specific tasks in that definition, the definition itself is really straightforward, and encapsulates the philosophy behind it. It's good to have a deeper explanation for maintainers to wrap their heads around, especially because we're also investing time in writing good descriptions for these tasks. For contributors, however, a higher-level description of the expectations they should have as coder and that we will have as reviewer/maintainer, would be faster to incorporate.

All of this is just to share an example for why I think separating the definition of "good first issue" for a primary audience of the contributor, rather than linking to the maintainer oriented definition, is important. I hope the example shows an effective way to layer this. I don't know if NeoMutt have deeper documentation elsewhere for their maintainers regarding this, but their docs are significantly more concise for new contributors than ours are to begin with. I think that's a good model, because if it's too long or looks too involved, people won't read it and fewer will be better off.

At least these three points (we don't have the mentorship ability) are true for us, and say things in a language a contributor would understand quickly:

> **Simple** – You don’t need to be a C master to do them
> **Understandable** – You don’t need to know the insides of NeoMutt
> **Low priority** – If it takes you a while, it doesn’t matter

For Openverse, maybe:

> **Simple** - You don't need to be a JavaScript or Python master to do them
> **Understandable** - Expectations are clearly documented in the issue and you don't need to learn everything about Openverse to complete them
> **Low priority** - If it takes you a while, it doesn't matter

I can't think of a better way to describe low priority :slightly_smiling_face: 

Another caveat for maintainers: good first issue shouldn't be part of a project stream at all, probably! Maybe only very small bugs or quality-of-life improvements. Unfortunately, we have very little feature related work that would be "good first issue" under that definition, though.

Reply:
I'm also realising here that NeoMutt explicitly says C _master_, implying, you need to know something about C to begin with. I wonder if we need to set that kind of bar as well. Like for each app, we would want to say something like:

- You should be comfortable with Python/TypeScript
- You should be familiar with the basic concepts for Django/Vue/Airflow
- etc.

Is the expectation that a good first issue is a way to learn about the technologies, or about Openverse? That's probably a good thing to clarify: what is our intention for someone completing good first issues? If we intend contributions to be focused on both, is there a baseline expectation for where someone is? Do we expect to need to leave feedback on a PR regarding the basics of JavaScript/Python (`is` vs `==` in Python, `==` vs `===` in JS, for example), or should contributors, even _first time_ contributors, already have that late-beginner/early-intermediate level of understanding about those languages (acknowledging there are sometimes complex edge cases that even advanced users of those languages struggle with)?

This is a bit of target-audience analysis that our issues could benefit from, and help narrow the scope of what details we need to include in a good first issue for them to be considered complete and appropriately detailed for the level of experience we anticipate and hope new Openverse contributors bring.

Reply:
For now I am going to remove this reference from this file. Your comments are fantastic so I'll make sure to reference them in a follow-up issue on documenting good first issues for _contributors_ rather than maintainers. 

Concerning this question:

> Is the expectation that a good first issue is a way to learn about the technologies, or about Openverse? 

I think the later, primarily learning about Openverse. Refining one's understanding of different technologies _can_ be an aspect of writing any code, but I do not think it is an explicit goal we need to promote. Ideally, issues are documented clearly enough that contributors can gauge if their current skillset is sufficient to cover the issue, or if they're nearly there and can fill any gaps on-the-fly. 

> Do we expect to need to leave feedback on a PR regarding the basics of JavaScript/Python (is vs == in Python, == vs === in JS, for example), or should contributors, even first time contributors, already have that late-beginner/early-intermediate level of understanding about those languages (acknowledging there are sometimes complex edge cases that even advanced users of those languages struggle with)?

IMO our contributors should have this level of late-beginner/early-intermediate knowledge. If they do not it's likely a PR will have significant issues and either be closed or taken over by a maintainer. I don't think we should spend time walking people through these types of learnings.

Reply:
Sounds good, a separate issue makes sense. I share your opinion on what appropriate expectations would be for our issues, and in particular think it aligns well with the time and focus we are able to dedicate and explicitly commit to for any PR review or contributions in general.